workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectDispatcherSystem.cs

```diff
@@ -175,18 +165,30 @@ public ActivationManagerNotFoundException(string message) : base(message)
             }
         }
 
-        public void CreateActivationManagerAndReaderWriterStore(Entity entity)
+        internal void CreateActivationManagerAndReaderWriterStore(Entity entity, GameObject gameObject)
         {
             if (entityToActivationManager.ContainsKey(entity))
             {
                 throw new ActivationManagerAlreadyExistsException($"MonoBehaviourActivationManager already exists for entity {entity.Index}.");
             }
 
-            var gameObject = EntityManager.GetComponentObject<GameObjectReference>(entity).GameObject;
             var store = new InjectableStore();
             entityToReaderWriterStore.Add(entity, store);
             var manager = new MonoBehaviourActivationManager(gameObject, injector, store, logger);
             entityToActivationManager.Add(entity, manager);
         }
+
+        internal void RemoveActivationManagerAndReaderWriterStore(Entity entity)
+        {
+            if (!entityToActivationManager.TryGetValue(entity, out var activationManager))
+            {
+                throw new ActivationManagerNotFoundException($"MonoBehaviourActivationManager not found for entity {entity.Index}.");
+            }
+
+            // Disable enabled SpatialOSBehaviours and dispose leftover Requirables.
+            activationManager.Dispose();
```

Comment:
Dispose this after removing the references to it.

Reply:
Also if we have disposable objects we need to have a OnDeleteManager function.
This system probably needs one anyway.